psi4/src/psi4/libfock/DFJCOSK.cc

```diff
@@ -73,19 +80,28 @@ Matrix compute_numeric_overlap(const DFTGrid &grid, const std::shared_ptr<BasisS
         bf_computer.compute_functions(block);
         auto point_values = bf_computer.basis_values()["PHI"];
 
+        // lambda for returning sign of double
+        auto sign = [ ](double val) {
```

Comment:
I do not like this: the return type should be obvious from the function definition.

It looks like the function returns an `int`.

Reply:
Yeah, on second glance, you're right, it does return an `int`. We can definitely clear up the return type; and likely, we want it to return a `double`, to boot.

These goals could be achieved pretty easily using one of two methods:
1. Place a `static_cast<double>` around the returned value in the lambda function.
2. Simply replace the current lambda function implementation with one using a ternary operator.

I will fix this up with idea #1 to start, as the entire current implementation avoids calling the equivalent of an `if` statement for each function call; but if people prefer using a ternary operator for clarity, I'll be happy to do that, as well.

Reply:
Yes a ternary would be preferable

Reply:
All right, I'll use a ternary instead, then!

Reply:
Done and done!